python/src/scipp/plot/plot_2d.py

```diff
@@ -282,20 +339,84 @@ def update_slice(self, change):
             if not val.disabled:
                 self.lab[dim].value = self.make_slider_label(
                     self.slider_x[self.name][dim], val.value)
-                vslice = vslice[val.dim, val.value]
+                dslice = dslice[val.dim, val.value]
                 # At this point, after masks were combined, all their
                 # dimensions should be contained in the data_array.dims.
                 if self.params["masks"][self.name]["show"]:
                     mslice = mslice[val.dim, val.value]
             else:
-                button_dims[self.buttons[dim].value.lower() == "x"] = val.dim
+                # Get the dimensions of the dimension-coordinates, since
+                # buttons can contain non-dimension coordinates
+                button_dims[self.buttons[dim].value.lower() ==
+                            "x"] = self.slider_x[self.name][val.dim].dims[0]
```

Comment:
Which buttons is the referring to? Aren't buttons labels with dimensions, not coord names?

Reply:
No the buttons are using the coordinate names: https://scipp.github.io/visualization/plotting-2d-data.html#Using-labels-along-some-axis

Reply:
I mean the buttons used to flip the X-Y axes of the image

Reply:
But that is something quite different: Labels are used for labeling the tics, but the dimension is the dimension, so buttons should maybe still be labeled with the dim?

Reply:
i think it would be counter-intuitive to ask for `axes=[y, somelabels]` and get a figure with `somelabels` as the x axis label, and then two buttons with `y` and `x` for their description. I think it would no be clear where the `x` came from.

Reply:
But aren't we actually using the dimension coord for plotting the data, and the axes just for the labeling?